ports/libsimpleservo/api/src/lib.rs

```diff
@@ -459,6 +463,12 @@ impl ServoGlue {
         self.process_event(WindowEvent::Keyboard(key_event))
     }
 
+    pub fn media_session_action(&mut self, action: i32) -> Result<(), &'static str> {
+        info!("Media session action {:?}", action);
+        let browser_id = self.get_browser_id()?;
```

Comment:
One potential issue here I can think of is that the `browser_id` will be the "active tab", whereas the "active media session" could, I think, be a tab playing media while not being the "active tab".

Have you considered not tying the media session action to a top-level at this point, and simply forwarding it "as is" to the constellation. The constellation could then keep track of which session is "active", and in which event-loop the session is found(allowing for iframe to have an active session, for example), and then forwarding the action with the `BrowsingContextId` of the active session to the event-loop where it is found(I guess via the pipeline corresponding to the BC)?